angular-client/src/pages/charging-page/components/fault-display/fault-display.component.ts

```diff
@@ -12,37 +11,41 @@ export default class FaultDisplay {
   faults: { faultName: string; time: string }[] = [];
   resetButton = {
     onClick: () => {
-      this.faults = [];
+      this.faults.shift();
```

Comment:
what's this doing?

Reply:
that cycles through the faults every time pressed (deletes first element in array).

Reply:
This button should still reset all the faults, not do this. The shifting should happen automatically. Add a check to the fault adding function that calls this shift method if the number of faults is >= 50, then add in the new fault